docs/publishing/release-your-plugin-with-github-actions.md

```diff
@@ -21,81 +21,47 @@ The GitHub Action workflow was originally created and shared by [argentum](https
-           env:
-             GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-             VERSION: ${{ github.ref }}
-           with:
-             tag_name: ${{ github.ref }}
-             release_name: ${{ github.ref }}
-             draft: false
-             prerelease: false
-
-         - name: Upload zip file
-           id: upload-zip
-           uses: actions/upload-release-asset@v1
-           env:
-             GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-           with:
-             upload_url: ${{ steps.create_release.outputs.upload_url }}
-             asset_path: ./${{ env.PLUGIN_NAME }}.zip
-             asset_name: ${{ env.PLUGIN_NAME }}-${{ steps.build.outputs.tag_name }}.zip
-             asset_content_type: application/zip
-
-         - name: Upload main.js
-           id: upload-main
-           uses: actions/upload-release-asset@v1
-           env:
-             GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-           with:
-             upload_url: ${{ steps.create_release.outputs.upload_url }}
-             asset_path: ./main.js
-             asset_name: main.js
-             asset_content_type: text/javascript
-
-         - name: Upload manifest.json
-           id: upload-manifest
-           uses: actions/upload-release-asset@v1
-           env:
-             GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-           with:
-             upload_url: ${{ steps.create_release.outputs.upload_url }}
-             asset_path: ./manifest.json
-             asset_name: manifest.json
-             asset_content_type: application/json
-
-         - name: Upload styles.css
-           id: upload-css
-           uses: actions/upload-release-asset@v1
-           env:
-             GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-           with:
-             upload_url: ${{ steps.create_release.outputs.upload_url }}
-             asset_path: ./styles.css
-             asset_name: styles.css
-             asset_content_type: text/css
+      runs-on: ubuntu-latest
+
+      steps:
+        - uses: actions/checkout@v3
+          with:
+            fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+        - name: Use Node.js
+          uses: actions/setup-node@v1
+          with:
+            node-version: "14.x" # You might need to adjust this value to your own version
+
+        # Get the version number and put it in an environment file
+        - name: Get Version
+          id: version
+          run: |
+            echo "tag=$(git describe --tags $(git rev-list --tags --max-count=1))" >> $GITHUB_ENV
+
+        # Build the plugin
+        - name: Build
+          id: build
+          run: |
+            npm install
+            npm run build --if-present
+
+        # Package the required files into a zip
+        - name: Package
+          run: |
+            mkdir ${{ github.event.repository.name }}
+            cp main.js manifest.json styles.css README.md ${{ github.event.repository.name }}
```

Comment:
```suggestion
            cp main.js manifest.json styles.css README.md ${{ github.event.repository.name }}
```

This assumes that the developer uses the plugin id as the repository name. To be honest, I'm not sure whether Obsidian cares about the zip file?

Reply:
I don't think Obsidian cares about the zip file, either. I do think it's convenient to have it if someone wants to download the compiled artifacts and not the `.zip` of the source tree that GitHub gives by default.

I like the way that [advanced-tables](https://github.com/tgrosinger/advanced-tables-obsidian/blob/main/.github/workflows/release.yml) is doing it -- the plugin name is set in the `env` block toward the top.

I will push a commit in the near future which fixes this and several other things mentioned above.